samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt

```diff
@@ -0,0 +1,636 @@
+                                "x " + "${binding.fragmentCamera2ImageCaptureViewfinder.height}",
+                    )
+
+                    Log.d(TAG, "Selected Preview Size: $previewSize")
+                    binding.fragmentCamera2ImageCaptureViewfinder.setAspectRatio(
+                        previewSize?.width ?: SIZE_720P.size.width,
+                        previewSize?.height ?: SIZE_720P.size.height,
+                    )
+                }
+            },
+        )
+
+        // Used to rotate the output media to match device orientation
+        relativeOrientation = OrientationLiveData(requireContext(), characteristics).apply {
+            observe(viewLifecycleOwner) { orientation ->
+                Log.d(TAG, "Orientation changed: $orientation")
+            }
+        }
+    }
+
+    private fun setUpImageReader() {
+        // Initialize an image reader which will be used to capture still photos
+        val configMap = characteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP)
+        configMap?.let { config ->
+            config.getOutputSizes(ImageFormat.JPEG).maxByOrNull { it.height * it.width }
+                ?.let { size ->
+                    imageReader = ImageReader.newInstance(
+                        size.width, size.height, ImageFormat.JPEG, IMAGE_BUFFER_SIZE,
+                    )
+                }
+        }
+
+        // Listen to the capture button
+        binding.fragmentCamera2ImageCaptureButton.setOnClickListener {
+            // Disable click listener to prevent multiple requests simultaneously in flight
+            it.isEnabled = false
+
+            // Perform I/O heavy operations in a different scope
+            lifecycleScope.launch(Dispatchers.IO) {
+                takePhoto().use { result ->
+                    Log.d(TAG, "Result received: $result")
+
+                    // Save the result to disk
+                    val output = saveResult(result)
+                    Log.d(TAG, "Image saved: ${output.absolutePath}")
+
+                    // If the result is a JPEG file, update EXIF metadata with orientation info
+                    if (output.extension == "jpg") {
+                        val exif = ExifInterface(output.absolutePath)
+                        exif.setAttribute(
+                            ExifInterface.TAG_ORIENTATION, result.orientation.toString(),
+                        )
+                        exif.saveAttributes()
+                        Log.d(TAG, "EXIF metadata saved: ${output.absolutePath}")
+                    }
+
+                    // Display the photo taken to user
+                    lifecycleScope.launch(Dispatchers.Main) {
+                        binding.fragmentCamera2ImageCaptureViewfinder.visibility = View.GONE
+                        binding.fragmentCamera2ImageCaptureButton.visibility = View.GONE
+
+                        parentFragmentManager.commit {
```

Comment:
The simple image viewer is already part of this fragment. You don't need the commit right?